### DISCUSSION TITLE
Thick Bridges option in the 2.4.1 merge

### DISCUSSION THREAD

**Discussion by neophyl** [supermerill/SuperSlicer#2379] [General]
The one option in PS2.4 that I use more than anything is to untick the 'Thick Bridges' option.  Are we going to get that option ported over to the merge too ?  

This option combined with the much improved supports has given me loads better lower surfaces when printing on top of supports.

**Comment by supermerill** [comment]
I'll add a widget to configure the settings to replicate it.

**Comment by tome9111991** [comment]
you can play with bride_overlap in Width & flow tab
![Screenshot 2022-02-14 220938](https://user-images.githubusercontent.com/57866234/153947058-f6894163-f8c0-450d-b046-06e9f66f3849.png)

I think it's similar to Thick Bridges

**Reply by supermerill** [reply]
You have to reduce the bridge flow and the density, but yes.
That's why I will create a scripted widget for that, to be easier for people who want this option.

**Reply by neophyl** [reply]
While I understand the original theory behind the assumption and while logic would say you are right I n practice it isn’t the same for real prints, as the bridge lines are just printed with the assumption that they aren’t nozzle diameter any more. To get a similar cross section you have to reduce it by a lot and then you get gaps between the lines. The resulting surface on both perimeters that are classed as an overhang and also of actual bridges, especially when printed on support is night and day. 
With Susi I always used to adjust the overhang thresholds to make perimeters not be classed as overhang and that gave me better perimeter results than PS on many models but still didn’t help with flat surface bridge areas, both PS and Susi were lacking in that area. 

The disabling of the logic in pS 2.4 made a dramatic difference. With it the perimeters are still overhang so do still kick in the cooling settings while still having normal perimeters cross sections which has improved things there, and flat surfaces on top of support have been worlds better. Also bridge infill uses the same logic. 

The new snug supports and that single tickbox for turning off thick bridges are the 2 best changes in 2.4 in my opinion.   I’ve updated every single one of my saved PS profiles to be that way since it was available in the 2.4 alphas. I really can’t think of a single reason for thick bridges after seeing the results of them turned off.

**Reply by supermerill** [reply]
I don't really understand. What is the difference ( for a given surface) between 
* 10 rectilinear bridges lines of 0.08mm3/mm, for a length of 10mm
* 10 rectilinear "not thick bridges" lines of 0.08mm3/mm, for a length of 10mm
?

**Comment by neophyl** [comment]
I don’t know the math behind it, I just know the results I’ve been getting. You give me some exact values to plug in what should be charged to duplicate the thin bridging flow and I’ll print some test pieces using whatever version of Susi you want.

**Comment by andilee7** [comment]
I also stumbled upon this issue. Sinces weeks I'm struggling about bad top layer quality. I thought long time I have some hardware issues. But it happens only on larger top surface areas which are printed over infill. It took me some time to figure it out because in general the top surface quality in SuSl is superior. I then found out, that I never printed such parts with a larger top area since I switched from Cura to SuSl.
However, after I compared my old Cura profile and made some test prints I quickly discovered that the reason is the very first layer on top of the infill. That layer is so overextruded, that it cannot compensate within the next 3 or 4 layers. I've not tried more because I think 5 top layers is enough.
I've played with print speeds, fan speed and bridge flow but I never got near the result of Cura.

**Cura - First and second layer on infill**
![cura_first_layer_over_infill](https://user-images.githubusercontent.com/95607655/158481506-e4d984c3-9106-4700-97ab-affbc090f047.jpg)
It's not sagging and also not bulging up. It's just a perfect bridged layer in my opinion. 
This results in following top layer (4 total top layers)
![cura_top](https://user-images.githubusercontent.com/95607655/158481640-e4353334-357d-49b5-b1db-cc09e782c143.jpg)

**SuperSlicer - First layer on infill**
![susl_first_layer_over_infill](https://user-images.githubusercontent.com/95607655/158481666-04198c72-5daf-448e-8029-5ea2036a3b0e.jpg)
That one is completely overextruded which is unable to recover for a good surface quality as seen here with also 4 top layers in total
![susl_top](https://user-images.githubusercontent.com/95607655/158481751-f197b454-f3e7-4d91-8bf4-fb02ef9c3ff6.jpg)

I also tried turning down the bridge flow to 25% and printing 5 top layers. Still not looking really good.

So I'm happy that the thick layer feature of PS 2.4 should address this issue (not tried it yet) and can't wait to get a stable SuperSlicer version based on 2.4 :)


By the way @supermerill: I would like to thank you for your hard work on SuperSlicer. I absolutly love it and never want look back to Cura since I first tried it.

**Reply by supermerill** [reply]
It's not a silver bullet. It just change the bridge flow & bridge spacing. You can already find your sweet spot.

**Reply by andilee7** [reply]
Does that mean it is not the same as PS 2.4 Thick Bridges? And your own implementation sets bridge_overlap, bridge_overlap_min and bridge_flow_ration, correct?
If that's the case I should be able to try it manually setting the values with the current stable release version. Do you have any recommended values to replicate the Cura behavior (and probably PS2.4)?

**Reply by supermerill** [reply]
> Does that mean it is not the same as PS 2.4 Thick Bridges?

The end result is the same. But as I added more settings to have more control over the bridges, I don't need that "kill switch".
for a 0.2mm layer on 0.4 mm nozzle, it's ~60% bridge_flow_ration and ~78% bridge_overlap.

Also, note that you may want ot set the fan speed & the bridge speed to the same value as the ones your cura use (or disable the bridge fan speed & set the speed the same as the solid infill if there is no settings for bridges)
